### PR TITLE
Add extra documentation around glue

### PR DESCRIPTION
**Pull request by jwoudenberg**
Adding some documentation around Glue following a conversation in Zulip:
https://roc.zulipchat.com/#narrow/channel/304641-ideas/topic/Platform.20Cookbook

Please don't hesitate to correct me where I'm wrong. I'm very new to this, so probably made a couple of mistakes!

### CODE REVIEW

**Review by lukewilliamboswell — APPROVED** [review]
This is helpful. Thank you

**Comment by jwoudenberg** [comment]
I think the failing test might be f flake? It runs for me locally, and I don't think it's related to these changes.

**Review by tannerntannern — COMMENTED** [review]
Just noting a few typos.  Sorry I didn't have a chance to comment before merging

File: crates/glue/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Glue
+
+Glue is a bit of tooling built into Roc to help with platform development. Roc platforms are written in a different language than Roc, and some it requires some finesse to let other languages to read and write Roc types like records and unions in a way compatible with how Roc uses those types.
```
[the remark this answers is not in the document]
Comment (lishaduck) [reply crates/glue/README.md:3]: You missed another typo :)
The second to in "to let other languages to read" should be dropped.

**Pull request merged**